superset/databases/commands/create.py

```diff
@@ -50,9 +50,13 @@ def run(self) -> Model:
 
             try:
                 TestConnectionDatabaseCommand(self._actor, self._properties).run()
-            except Exception:
-                db.session.rollback()
-                raise DatabaseConnectionFailedError()
+            except Exception:  # pylint: disable=broad-except
+                with event_logger(
+                    action="db_connection_failed",
+                    engine=database.db_engine_spec.__name__,
+                ):
+                    db.session.rollback()
+                    raise DatabaseConnectionFailedError()
```

Comment:
For places where we do a rollback we should it **outside** the context manager, otherwise if the logging fails the transaction won't be rolled back:

```suggestion
                db.session.rollback()
                event_logger.log_with_context(
                    action="db_connection_failed",
                    engine=database.db_engine_spec.__name__,
                )
                raise DatabaseConnectionFailedError()
```

Reversing the order ensures that even if the logging fails the rollback will still go through.